### PR TITLE
[+] BO CSV: new Export CSV for new Product catalog admin page.

### PR DESCRIPTION
**Pull request by xGouley**

### CODE REVIEW

File: Adapter/Product/AdminProductDataProvider.php [hunk 1]
```diff
@@ -293,12 +295,14 @@ public function getCatalogProductList($offset, $limit, $orderBy, $sortOrder, $po
...
         $currency = new \CurrencyCore(\Configuration::get('PS_CURRENCY_DEFAULT'));
         foreach ($products as &$product) {
-            $product['price'] = \ToolsCore::displayPrice($product['price'], $currency);
             $product['total'] = $total; // total product count (filtered)
             $product['price_final'] = \ProductCore::getPriceStatic($product['id_product'], true, null,
-                    (int)\Configuration::get('PS_PRICE_DISPLAY_PRECISION'), null, false, true, 1,
-                    true, null, null, null, $nothing, true, true);
-            $product['price_final'] = \ToolsCore::displayPrice($product['price_final'], $currency);
+                (int)\Configuration::get('PS_PRICE_DISPLAY_PRECISION'), null, false, true, 1,
+                true, null, null, null, $nothing, true, true);
```
Comment (mickaelandrieu) [inline Adapter/Product/AdminProductDataProvider.php]: $nothing is undefined or global, right ?

File: src/PrestaShopBundle/Controller/Admin/ProductController.php [hunk 2]
```diff
@@ -674,6 +674,45 @@ public function unitAction(Request $request, $action, $id)
         }
     }
 
+    public function exportAction($_format)
+    {
+        // init vars
+        $productProvider = $this->container->get('prestashop.core.admin.data_provider.product_interface');
+        /* @var $productProvider ProductInterfaceProvider */
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: Can you reverse the PHPDoc annotation and the variable declaration ?

```php
/* @var $productProvider ProductInterfaceProvider */
$productProvider = $this->container->get('prestashop.core.admin.data_provider.product_interface');
```

I'm not sure the parsed information is valid this way...

File: Adapter/Product/AdminProductDataProvider.php (hunk 1 above)
Comment (xGouley) [inline Adapter/Product/AdminProductDataProvider.php]: $nothing is undefined. Mandatory due to legacy code that we cannot change for our Adapter needs.

File: src/PrestaShopBundle/Controller/Admin/ProductController.php (hunk 2 above)
Comment (xGouley) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: The annotation works in Eclipse and PHPStorm in this way. If there is documentation on the official syntax, I will follow it.

Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: I found this sample on phpdocumentor docs (http://manual.phpdoc.org/HTMLSmartyConverter/PHP/phpDocumentor/tutorial_tags.var.pkg.html), still not official docs to me :)

Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: Ok, thanks for information

**Pull request merged**